### ISSUE TITLE
UnnecessaryBooleanExpression: false positive on Spock tests with ||

### ISSUE THREAD

**Issue by php-coder** [CodeNarc/CodeNarc#176]
Spock allows to use `||` as a separator between input data and result in `where` block: http://spockframework.org/spock/docs/1.1-rc-2/data_driven_testing.html#_syntactic_variations But CodeNarc reports about `UnnecessaryBooleanExpression` violation in this case.

**Comment by chrismair** [comment]
In case you are not aware, if your Spock classes follow some kind of naming convention or are in a separate folder, you could turn off that rule for those classes as a workaround, e.g.
e.g. set `doNotApplyToClassNames = "*Specification"` for UnnecessaryBooleanExpression

As we've encountered with some other Spock-related CodeNarc issues, there seems to be no universal naming convention or good way for the rules to determine that they are within a Spock specification.

**Comment by fistach** [comment]
Hello Chris.

Can this rule checker detect that it encountered the 'where:' block? It is only there, where we would like to skip this rule. Skipping the rule for the whole *Specification is very broad. In teams doing TDD this would be the most of their code.

**Comment by chrismair** [comment]
From what I can see in the Groovy AST, the label is only associated with a single (the first) statement under it. I think the logic would have to be more involved to track all statements following that label but before any other one.

I am hesitant to add complex Spock-specific logic to a rule, impacting the many non-Spock users. But if it can be straightforward, then that would probably be worth doing.

**Comment by vanta** [comment]
@chrismair any update here? Fix or close? :) It's really annoying if you're are the Spock user.

**Comment by chrismair** [comment]
There are no plans to fix that at this point. I think the recommendation would be to disable that one rule for test classes/files.

**Comment by albertocavalcante** [comment]
Just checking to see if ignoring this rule for SpockTest remains the official guidance. Thanks.

**Comment by chrismair** [comment]
Yes, that is currently the recommendation for Spock test classes/files.
